app/controllers/labware_controller.rb

```diff
@@ -57,7 +57,11 @@ def history
   def update # rubocop:todo Metrics/AbcSize, Metrics/MethodLength
     respond_to do |format|
       params_hash = params.to_unsafe_h
+      current_retention_instruction = @asset.retention_instruction
       if @asset.update(labware_params.merge(params_hash.fetch(:lane, {})))
+        if params_hash[:labware].key?(:retention_instruction)
+          EventFactory.record_retention_instruction_updates(@asset, current_user, current_retention_instruction)
+        end
```

Comment:
I presume this `update` method is called by multiple pages - at least the normal 'edit' page, and your new 'retention instructions' page?

So is this `params_hash[:labware].key?(:retention_instruction)` basically checking if you're on the retention instructions page ... by checking if the `params` that come from the page include the retention_instruction?

Would an alternative be to check `@asset.retention_instruction` again (because you've just called `update` so it should have changed), and check if it's different to `current_retention_instruction`? Not sure if it's better or worse - I guess possibly more future proof because it doesn't rely on the name of the key in the params hash?

Reply:
Yeah it is better. I've updated it :)